infogami/infobase/account.py

```diff
@@ -197,16 +199,23 @@ def find_account(self, email=None, username=None):
             return store.get(account_key)
 
     def set_auth_token(self, user_key):
-        t = datetime.datetime(*time.gmtime()[:6]).isoformat()
+        t = datetime(*time.gmtime()[:6]).isoformat()
         text = "%s,%s" % (user_key, t)
         text += "," + self._generate_salted_hash(self.secret_key, text)
         web.ctx.infobase_auth_token = text
 
     #### Utilities
 
     def _generate_salted_hash(self, key, text, salt=None):
-        salt = salt or hmac.HMAC(key, str(random.random())).hexdigest()[:5]
-        hash = hmac.HMAC(key, web.safestr(salt) + web.safestr(text)).hexdigest()
+        """
```

Comment:
Especially since we're not using Open Library for passwords, I am less worried about this block, but it's something we should consider keeping an eye on re: tests